Source/Diagnostics/WarpXOpenPMD.cpp

```diff
@@ -390,6 +386,46 @@ WarpXOpenPMDPlot::DumpToFile (WarpXParticleContainer* pc,
          {
            // Save positions
            auto const positionComponents = detail::getParticlePositionComponentLabels();
+#if defined(WARPX_DIM_RZ)
+           {
+              std::shared_ptr<amrex::ParticleReal> z(
+                      new amrex::ParticleReal[numParticleOnTile],
+                      [](amrex::ParticleReal const *p) { delete[] p; }
+              );
+              for (auto i = 0; i < numParticleOnTile; i++)
+                  z.get()[i] = aos[i].m_rdata.pos[1];  // {0: "r", 1: "z"}
+              std::string const positionComponent = "z";
+              currSpecies["position"]["z"].storeChunk(z, {offset}, {numParticleOnTile64});
+           }
+
+           //   reconstruct x and y from polar coordinates r, theta
+           amrex::ParticleReal const* theta = nullptr;
+           for (auto idx=0; idx<m_NumSoARealAttributes; idx++) {
```

Comment:
Is there a reason you have this search instead of using PIdx::theta?

Reply:
Thanks for the hint, I was not aware of that. Will simplify :)